src/poetry/utils/exporter.py

```diff
@@ -174,9 +177,11 @@ def _export_requirements_txt(
 
     def _output(self, content: str, cwd: Path, output: IO | str) -> None:
         decoded = decode(content)
-        try:
-            output.write(decoded)
-        except AttributeError:
+        if isinstance(output, str):
             filepath = cwd / output
             with filepath.open("w", encoding="utf-8") as f:
                 f.write(decoded)
+            return
+        with contextlib.suppress(AttributeError):
+            # FIXME
+            output.write(decoded)
```

Comment:
I imagine we don't need to worry about `AttributeError` and can just rely on the instance check:

```suggestion
        output.write(decoded)
```